R/spread.R

```diff
@@ -0,0 +1,27 @@
+#' @export
+spread.tbl_sql <- function(data, key, value, fill = NA){
+  key <- enexpr(key)
+  value <- enexpr(value)
+  fill <- enexpr(fill)
+
+  # Obtains unique values
+  keys <- data %>%
+    group_by(!! key) %>%
+    summarise() %>%
+    pull()
+
+  # Builds an `ifelse()` statement per unique value
+  conds <- keys %>%
+    purrr::map(~expr(ifelse((!! key) == (!! .x), !! value, (!! fill))))
+
+  # Sets names
+  nf <- keys %>%
+    purrr::map(~expr(!! sym(.x)))
+
+  # Adds a mutate per each unique value
+  for(vals in 1:length(keys)){
+    data <- mutate(data, !! expr_text(nf[[vals]]) := !! conds[[vals]])
```

Comment:
Conceptually I think the operation should be a summarise() instead of mutate(). You can see your result differs from tidyr::spread() if you use the following test data.

```r
df <- tibble(
  h = c("a", "b", "c", "d", "a", "b", "c", "d"),
  x = c("A", "A", "B", "B", "B", "B", "A", "A"),
  y = c("X", "Y", "X", "Y", "X", "Y", "X", "Y"),
  z = c( 1,   2,   3,   4,   5,   6,   7,   8 )
)
```